app/src/services/read-only.service.js

```diff
@@ -0,0 +1,29 @@
+module.exports = class ReadOnlyService {
+
+    constructor(blacklist = [], whitelist = []) {
+        this.blacklist = blacklist;
+        this.whitelist = whitelist;
+    }
+
+    isBlacklisted(path) {
+        return this.blacklist.some((element) => new RegExp(element).test(path));
+    }
+
+    isWhitelisted(path) {
+        return this.whitelist.some((element) => new RegExp(element).test(path));
+    }
+
+    shouldBlockRequest(method, path) {
+        return (ReadOnlyService.isRead(method) && this.isBlacklisted(path))
+        || (ReadOnlyService.isWrite(method) && !this.isWhitelisted(path));
+    }
+
+    static isRead(method) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

Reply:
Added OPTIONS and HEAD methods as supported read methods :+1: